spec/creator_spec.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+require 'creator'
+
+RSpec.describe Creator do
+  describe '#run' do
+    let(:database) { double('ArrayDatabase', create: nil) }
+    let(:ui) do
+      double('UserInterface',
+             ask_for_fields: test_details,
+             display: nil,
+             add_another_contact?: false,
+             menu_choice: nil)
+    end
+
+    it 'tells the UI to ask for contact details' do
+      creator = described_class.new(ui, database)
+
+      creator.run
+
+      expect(ui).to have_received(:ask_for_fields).once
+    end
+
+    it 'tells the database to add the contact returned by the UI' do
+      creator = described_class.new(ui, database)
+
+      creator.run
+
+      expect(database).to have_received(:create).with(test_details)
+    end
+
+    it 'tells the user interface to display the contact that was added' do
+      creator = described_class.new(ui, database)
+
+      creator.run
+
+      expect(ui).to have_received(:display).with(test_details)
+    end
+
+    it 'tells the user interface to ask the user if they want to add another contact' do
+      creator = described_class.new(ui, database)
+
+      creator.run
+
+      expect(ui).to have_received(:add_another_contact?).once
+    end
+
+    it 'repeats the previous steps if the user wants to add another contact' do
+      allow(ui).to receive(:add_another_contact?).and_return(true, false)
+
+      creator = described_class.new(ui, database)
+
+      creator.run
+
+      expect(ui).to have_received(:ask_for_fields).twice
+      expect(database).to have_received(:create).twice
+      expect(ui).to have_received(:display).twice
+      expect(ui).to have_received(:add_another_contact?).twice
```

Comment:
It's a good practice to have only 1 `expect` per test, except in very extremely edge cases.

In this case the most important thing that happens is that you add two contacts to the database, so I would write an expectation for that :+1:

Reply:
How can I test that create has been called if it is not a double? 
The only way I can think of is to test an actual ArrayDatabase's array for the contact, but wouldn't I be testing the ArrayDatabase class if I did that?

Reply:
Would it be miss leading to test the ArrayDatabase class in the Creator class spec?

Reply:
```ruby
expect(database.count).to eq(2)
```

Reply:
So, it is ok to test other class' as long as their in the chain of the described class?